lib/collection/property-list.js

```diff
@@ -360,12 +360,47 @@ _.assign(PropertyList.prototype, /** @lends PropertyList.prototype */ {
      * Check whether an item exists in this list
      *
      * @param {String|PropertyList.Type} item
+     * @param value
      * @returns {Boolean}
      */
-    has: function (item) {
-        return Boolean(_.isString(item) ? this.one(item) : this.filter(function (member) {
-            return member === item;
-        }).length);
+    has: function (item, value) {
+        var match,
+            val,
+            i;
+
+        match = _.isString(item) ?
+            this.reference[this._postman_listIndexCaseInsensitive ? item.toLowerCase() : item] :
+            this.filter(function (member) {
+                return member === item;
+            });
+
+        // If we don't have a match, there's nothing to do
+        if (!match) { return false; }
+
+        // if no value is provided, just check if item exists
+        if (arguments.length === 1) {
+            return Boolean(_.isArray(match) ? match.length : match);
+        }
+
+        // If this property allows multiple values and we get an array, we need to iterate through it and see
+        // if any element matches.
+        if (this._postman_propertyAllowsMultipleValues && _.isArray(match)) {
+            for (i = 0; i < match.length; i++) {
```

Comment:
For loop is faster, and this is a lower level function, so we'll not introduce extra function calls.